pandas/_libs/window/aggregations.pyx

```diff
@@ -502,9 +531,20 @@ def roll_skew(ndarray[float64_t] values, ndarray[int64_t] start,
         start, end
     )
     output = np.empty(N, dtype=float)
+    min_val = np.nanmin(values)
 
     with nogil:
+        for i in range(0, N):
+            val = values[i]
+            if notnan(val):
+                nobs_mean += 1
+                sum_val += val
+    mean_val = sum_val / nobs_mean
+    # Other cases would lead to imprecision for smallest values
+    if min_val - mean_val > -1e5:
+        values = values - round(mean_val)
 
+    with nogil:
```

Comment:
you can leave the original nogil right? e.g. no reason to do it twice?

Reply:
The subtraction of the mean does not wirk with nogil, that is the reason did it there again

Reply:
umm why? this is array - scalar right?

Reply:
Thought so too, but got an error. Will try again

Reply:
Hmmm, getting the following error without gil:

```
Error compiling Cython file:
------------------------------------------------------------
...
                nobs_mean += 1
                sum_val += val
        mean_val = sum_val / nobs_mean
        # Other cases would lead to imprecision for smallest values
        if min_val - mean_val > -1e5:
            values = values - round(mean_val)
                                   ^
------------------------------------------------------------

pandas/_libs/window/aggregations.pyx:545:36: Converting to Python object not allowed without gil
Traceback (most recent call last):
  File "setup.py", line 790, in <module>
    setup_package()
  File "setup.py", line 760, in setup_package
    ext_modules=maybe_cythonize(extensions, compiler_directives=directives),
  File "setup.py", line 539, in maybe_cythonize
    return cythonize(extensions, *args, **kwargs)
  File "/home/developer/anaconda3/envs/omi_reports/pandas-dev/lib/python3.8/site-packages/Cython/Build/Dependencies.py", line 1102, in cythonize
    cythonize_one(*args)
  File "/home/developer/anaconda3/envs/omi_reports/pandas-dev/lib/python3.8/site-packages/Cython/Build/Dependencies.py", line 1225, in cythonize_one
    raise CompileError(None, pyx_file)
```

Reply:
use np.round

Reply:
Same error unfortunately

```
Error compiling Cython file:
------------------------------------------------------------
...
                nobs_mean += 1
                sum_val += val
        mean_val = sum_val / nobs_mean
        # Other cases would lead to imprecision for smallest values
        if min_val - mean_val > -1e5:
            values = values - np.round(mean_val)
                                      ^
------------------------------------------------------------

pandas/_libs/window/aggregations.pyx:545:39: Converting to Python object not allowed without gil
Traceback (most recent call last):
  File "setup.py", line 790, in <module>
    setup_package()
  File "setup.py", line 760, in setup_package
    ext_modules=maybe_cythonize(extensions, compiler_directives=directives),
  File "setup.py", line 539, in maybe_cythonize
    return cythonize(extensions, *args, **kwargs)
  File "/home/developer/anaconda3/envs/omi_reports/pandas-dev/lib/python3.8/site-packages/Cython/Build/Dependencies.py", line 1102, in cythonize
    cythonize_one(*args)
  File "/home/developer/anaconda3/envs/omi_reports/pandas-dev/lib/python3.8/site-packages/Cython/Build/Dependencies.py", line 1225, in cythonize_one
    raise CompileError(None, pyx_file)
Cython.Compiler.Errors.CompileError: pandas/_libs/window/aggregations.pyx
```

Reply:
maybe `cnp. PyArray_Round`?

Reply:
Hm no, mean_value is a float. This does not work without round either. Maybe the broadcasting is the problem here?

Reply:
yeah maybe have to use `np.broadcast_to`

Reply:
Found a solution for the round issue. Used a loop for the subtraction, because ``np.broadcast_to`` did not work either. No matter this, I think the loop is the most intuitive solution

Reply:
`from libc.math cimport round`?

Reply:
That works, thanks very much.